Compilation issue with catch.hpp

-----------------------------------------

If you have a question about the RDKit or aren't sure if what you're seeing is the right behavior please use the Discussions tab above (https://github.com/rdkit/rdkit/discussions) instead of posting it here.

Please help us help you: We normally need the information below in order to answer questions or understand bug reports. If you do not provide the information requested we may not be able to help you and will probably close the issue.

PLEASE DELETE THIS SECTION AFTER YOU HAVE READ IT

-----------------------------------------

**Describe the bug**
The compilation of **Release_2021_09_5** fails in Archlinux due to some issue related to `catch.hpp`

**To Reproduce**
Compile Release_2021_09_5 ( Note also Release_2021_09_4)  in an updated Archlinux using `Release_2021_09_5.tar.gz` and it will fail. This is the error you will get that halts the installation:

```CMake
[ 17%] Building CXX object Code/DataStructs/Wrap/CMakeFiles/cDataStructs.dir/wrap_FPB.cpp.o
In file included from /usr/include/signal.h:328,
                 from /home/hector/ownCloudUOCHB/repo/admin/arch/AUR/rdkit/src/rdkit-Release_2021_09_5/External/catch/catch/single_include/catch2/catch.hpp:8031,
                 from /home/hector/ownCloudUOCHB/repo/admin/arch/AUR/rdkit/src/rdkit-Release_2021_09_5/Code/Geometry/catch_tests.cpp:12:
/home/hector/ownCloudUOCHB/repo/admin/arch/AUR/rdkit/src/rdkit-Release_2021_09_5/External/catch/catch/single_include/catch2/catch.hpp:10815:58: error: call to non-‘constexpr’ function ‘long int sysconf(int)’
10815 |     static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
      |                                                          ^~~~~~~~~~~
In file included from /usr/include/bits/sigstksz.h:24,
                 from /usr/include/signal.h:328,
                 from /home/hector/ownCloudUOCHB/repo/admin/arch/AUR/rdkit/src/rdkit-Release_2021_09_5/External/catch/catch/single_include/catch2/catch.hpp:8031,
                 from /home/hector/ownCloudUOCHB/repo/admin/arch/AUR/rdkit/src/rdkit-Release_2021_09_5/Code/Geometry/catch_tests.cpp:12:
/usr/include/unistd.h:640:17: note: ‘long int sysconf(int)’ declared here
  640 | extern long int sysconf (int __name) __THROW;
      |                 ^~~~~~~
In file included from /home/hector/ownCloudUOCHB/repo/admin/arch/AUR/rdkit/src/rdkit-Release_2021_09_5/Code/Geometry/catch_tests.cpp:12:
/home/hector/ownCloudUOCHB/repo/admin/arch/AUR/rdkit/src/rdkit-Release_2021_09_5/External/catch/catch/single_include/catch2/catch.hpp:10874:45: error: size of array ‘altStackMem’ is not an integral constant-expression
10874 |     char FatalConditionHandler::altStackMem[sigStackSize] = {};
      |                                             ^~~~~~~~~~~~
```

**Expected behavior**
It should compile without error

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Configuration (please complete the following information):**
 - RDKit version:  Release_2021_09_5
 - OS: Archlinux (3/13/2022)
 - Python version (if relevant): 3.10.2
 - Are you using conda? No
 - If you are using conda, which channel did you install the rdkit from?
 - If you are not using conda: how did you install the RDKit?
-  GCC version: 11.2.0

**Additional context**
Add any other context about the problem here.

--- a/External/catch/CMakeLists.txt
+++ b/External/catch/CMakeLists.txt
@@ -12,10 +12,10 @@ elseif(EXISTS "${CATCH_DIR}/single_include/catch2/catch.hpp")
 else()
   message("-- Catch not found in ${CATCH_DIR}")
   include(RDKitUtils)
-  set(RELEASE_NO "2.12.4")
+  set(RELEASE_NO "2.13.8")
   downloadAndCheckMD5("https://github.com/catchorg/Catch2/archive/v${RELEASE_NO}.tar.gz"
         "${CMAKE_CURRENT_SOURCE_DIR}/v${RELEASE_NO}.tar.gz"
-        "dbe4fa691e37dd51256876641a821ec8")
+        "3639fb4cb8020de0dcf1fd5addb39418")
   execute_process(COMMAND ${CMAKE_COMMAND} -E tar zxf
     ${CMAKE_CURRENT_SOURCE_DIR}/v${RELEASE_NO}.tar.gz
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})

